packages/api/src/context.ts

```diff
@@ -0,0 +1,59 @@
+import { type inferAsyncReturnType } from "@trpc/server";
+import { type CreateNextContextOptions } from "@trpc/server/adapters/next";
+import {
+  type NodeHTTPCreateContextFnOptions,
+  type NodeHTTPRequest,
+  type NodeHTTPResponse,
+} from "@trpc/server/adapters/node-http";
+import jwt from "jsonwebtoken";
+
+import { prisma } from "@blobscan/db";
+
+import { SECRET } from "./env";
+
+type CreateContextOptions =
+  | NodeHTTPCreateContextFnOptions<NodeHTTPRequest, NodeHTTPResponse>
+  | CreateNextContextOptions;
+
+type CreateInnerContextOptions = Partial<CreateContextOptions> & {
+  apiClient: string | null;
+};
+
+function getJWTFromRequest(
+  req: NodeHTTPRequest | CreateNextContextOptions["req"],
+) {
+  const authHeader = req.headers.authorization;
+  if (!authHeader) {
+    return null;
+  }
+
+  const [type, token] = authHeader.split(" ");
+  if (type !== "Bearer" || !token) {
+    return null;
+  }
+
+  const decoded = jwt.verify(token, SECRET) as string;
+
+  return decoded;
+}
+
+export function createTRPCInnerContext(opts?: CreateInnerContextOptions) {
+  return {
+    prisma,
+    apiClient: opts?.apiClient,
+  };
+}
+
+export function createTRPCContext(opts: CreateContextOptions) {
+  const apiClient = getJWTFromRequest(opts.req);
+
+  const innerContext = createTRPCInnerContext({ apiClient });
+
+  return {
+    ...innerContext,
+    req: opts.req,
+    res: opts.res,
+  };
+}
+
+export type TRPCContext = inferAsyncReturnType<typeof createTRPCInnerContext>;
```

Comment:
This is how we get support to both API clients (express, web)?

Reply:
`inferAsyncReturnType` is just a little helper provided by trpc to get the context type itself.

Right now, the way we support both express and nextjs consumers is by doing [this](https://github.com/Blobscan/blobscan/blob/334c77ac65516b84854ee3884684107d3da075a4/packages/api/src/context.ts#L14).

As our procedures logic does not rely on any specific features of Express or Next.js, this approach has not caused any issues. However, we may need to reevaluate and consider adopting a more framework-agnostic strategy in the future.